app/src/test/java/com/mstudio/superheroarch/presentation/MainViewModelTest.kt

```diff
@@ -60,11 +61,12 @@ class MainViewModelTest {
 
     @Test
     fun `given main screen, when user clicks on dead filter, then show all dead characters only`() = runTest {
-        val expectedCharacters = listOf(RickAndMortyRepositoryInstruments.givenACharacterEntity().toCharacterData())
+        val expectedCharacters = listOf(RickAndMortyRepositoryInstruments.givenACharacterEntity(status = CharactersFilters.DEAD.type).toCharacterData())
         `when`(useCase.getAllCharacters()).thenReturn(expectedCharacters)
-        viewModel.onFilterButtonClicked(StatusFilters.DEAD)
+        viewModel.onCreate()
+        viewModel.onFilterButtonClicked(CharactersFilters.DEAD)
 
-        verify(view).showCharacters(expectedCharacters.filter { it.status == StatusFilters.DEAD.status })
+        verify(view, atLeastOnce()).showCharacters(expectedCharacters.filter { it.status == CharactersFilters.DEAD.type })
```

Comment:
atLeastOnce seems a bit lazy for a test, here the showCharacter should have been called twice, right? one for onCreate and another one for onFilterButtonClicked. Better to be specific, as if someone adds a bug that calls again the showCharacters, we won't realize using this test.